main.nf

```diff
@@ -54,7 +54,18 @@ if (params.max_cpus < 16 || params.max_memory < 30) {
         ------------------------------------
         ERROR: Insufficient resources: ${params.max_cpus} CPUs and ${params.max_memory} of memory.
```

Comment:
it seems you cited 'To run Mutect2 the pipeline requires at least 8 CPUs and 16 GB of memory', so I think the nested if block needs to be changed or updated depending on intended use.
```
if (params.max_cpus < 16 || params.max_memory < 30) {
    if (params.algorithm.contains('muse') || params.algorithm.contains('mutect2'))
```
currently I think if you specify mutect2 and provide the cited resources it will error.

I might suggest inverting these if blocks to check first for algorithm, and then check the resources meet the minimum requirement for the given algorithm.

```
if (params.algorithm.contains('muse') && (params.max_cpus < 16 || params.max_memory < 32)){...}
else if (params.algorithm.contains('mutect2') && (params.max_cpus < 8 || params.max_memory < 16)){...}
```

Reply:
Oh good catch @kiarod.  The first pair of cutoffs was supposed to be 8 and 16.  It was accidentally reverted when I merged in the `main` branch using `--no-ff`.  I need to learn a better way to do this.

Do you think it's worth inverting the blocks?  I think it would still require the same number of nested if statements and (trivially) would require evaluation of both every time.

Reply:
no worries! you're 100% right. Operationally they are the same thing, so your call on the if-blocks. The motivation for inverting the if-blocks was more so because the purpose of the if-blocks are to check sufficient resource requirement for specific algorithms, so in the case I suggested, there is one check per algorithm so it might be more straightforward to read/update/maintain, but no real performance difference especially considering these blocks are encountered once in a program that basically runs for a minimum of 2 hours haha

Reply:
I noticed only one line of the error message was being printed, so I've updated them.
See new error messages:
`/hot/code/sfitzgibbon/gitHub/uclahs-cds/pipeline-call-sSNV/slurm-82745.out`
`/hot/code/sfitzgibbon/gitHub/uclahs-cds/pipeline-call-sSNV/slurm-82746.out`